src/Sylius/Bundle/CoreBundle/Doctrine/Migrations/AbstractMySqlMigration.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ *  This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Doctrine\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+abstract class AbstractMySqlMigration extends AbstractMigration
+{
+    public function preUp(Schema $schema): void
+    {
+        $this->abortIf(!$this->isMySql(), 'Migration can only be executed safely on \'mysql\'.');
```

Comment:
are we sure, it will work with MariaDB?

Reply:
Yep. I do a `is_a()` check. In every version `MariaDBPlatform` (or `MariaDb1027Platform` in dbal2) extends `MySQLPlatform` (or `MySqlPlatform` in dbal2) so `$this->isMySql()` will return true for both `MySql` and `MariaDB`. Otherwise our MariaDB workflow would fail :P.